fuzzing/private/binary.bzl

```diff
@@ -0,0 +1,163 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Defines a rule for creating an instrumented fuzzing executable."""
+
+load("//fuzzing/private:engine.bzl", "CcFuzzingEngineInfo")
+load(
+    "//fuzzing/private:instrum_opts.bzl",
+    "instrum_defaults",
+    "instrum_opts",
+)
+load(
+    "//fuzzing:instrum_opts.bzl",
+    "instrum_configs",
+    "sanitizer_configs",
+)
+
+CcFuzzingBinaryInfo = provider(
+    doc = """
+Provider for storing information about a fuzz test binary.
+""",
+    fields = {
+        "binary_file": "The instrumented fuzz test executable.",
+        "binary_runfiles": "The runfiles of the fuzz test executable.",
+        "corpus_dir": "The directory of the corpus files used as input seeds.",
+        "dictionary_file": "The dictionary file to use in fuzzing runs.",
+        "engine_info": "The `CcFuzzingEngineInfo` provider of the fuzzing engine used in the fuzz test.",
+    },
+)
+
+def _fuzzing_binary_transition_impl(settings, attr):
```

Comment:
is `attr` ever used?

Reply:
It is not for this particular transition, but this is part of the standard signature required by the `implementation` attribute of the `transition()` definition below. (It's a bit like `argc` and `argv` in a main function.)